### PR TITLE
Feature/filter

### PR DESCRIPTION
**Pull request by WorldFam**

### CODE REVIEW

File: app/app/src/main/java/com/warehouse/data/Room/Statistics.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    public String getRoomId() {
+        return roomId;
+    }
+
+    public void setRoomId(String roomId) {
+        this.roomId = roomId;
+    }
+
+    public String getStringId() {
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/data/Room/Statistics.java:28]: getSensorId

File: app/app/src/main/java/com/warehouse/data/Room/Statistics.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return sensorId;
+    }
+
+    public void setStringId(String stringId) {
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/data/Room/Statistics.java:32]: setSensorId

File: app/app/src/main/java/com/warehouse/services/ServiceGenerator.java [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
 
 import com.warehouse.data.Room.RoomApi;
+import com.warehouse.data.Room.StatisticsApi;
 
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
 
 public class ServiceGenerator {
-    private static Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
+    private static final Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/services/ServiceGenerator.java:10]: do we need to make it final?

File: app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java [hunk 4]
```diff
@@ -97,19 +109,19 @@ private void setupTemperatureChart() {
...
+//        entries.add(new Entry(3, 2300));
+//        entries.add(new Entry(4, 1200));
+//        entries.add(new Entry(5, 5400));
+//        entries.add(new Entry(6, 9100));
+//        entries.add(new Entry(7, 2500));
 
 
         // Data set
-        LineDataSet dataSet = new LineDataSet(entries, getResources().getString(R.string.chart_temperature));
+        LineDataSet dataSet = new LineDataSet(getTemperatureEntries (), getResources().getString(R.string.chart_temperature));
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java:124]: I think it should work a little bit different

1. Observe the the dastboardViewModel.getActivity
2. transform the observed value
3. set the data set with the recently transformed value

For example:

```
dashboardViewModel.getActivity ("temperature").observe(new Observe..... {
         (values) => {
              ArrayList<Entry> entries = new ArrayList<>();
              Iterator iterator = values.entrySet().iterator();
              
              ...set the entries values
              
              LineDataSet dataSet = new LineDataSet(entries (), getResources().getString(R.string.chart_temperature));
              ...update the chart with the new dataSet
})
```

File: app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java [hunk 5]
```diff
@@ -144,19 +169,19 @@ private void setupHumidityChart() {
...
+//        entries.add(new Entry(3, 2300));
+//        entries.add(new Entry(4, 1200));
+//        entries.add(new Entry(5, 5400));
+//        entries.add(new Entry(6, 9100));
+//        entries.add(new Entry(7, 2500));
 
 
         // Data set
-        LineDataSet dataSet = new LineDataSet(entries, getResources().getString(R.string.chart_temperature));
+        LineDataSet dataSet = new LineDataSet(getHumidityEntries (), getResources().getString(R.string.chart_temperature));
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java:184]: same here

File: app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java [hunk 6]
```diff
@@ -191,19 +229,19 @@ private void setupCO2Chart() {
...
+//        entries.add(new BarEntry(3, 2300));
+//        entries.add(new BarEntry(4, 1200));
+//        entries.add(new BarEntry(5, 5400));
+//        entries.add(new BarEntry(6, 9100));
+//        entries.add(new BarEntry(7, 2500));
 
 
         // Data set
-        BarDataSet dataSet = new BarDataSet(entries, getResources().getString(R.string.chart_temperature));
+        BarDataSet dataSet = new BarDataSet(getCo2Entries (), getResources().getString(R.string.chart_temperature));
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java:244]: same here

File: app/app/src/main/java/com/warehouse/services/ServiceGenerator.java (hunk 3 above)
Comment (WorldFam) [reply app/app/src/main/java/com/warehouse/services/ServiceGenerator.java:10]: It is was recommended 💡

File: app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java [hunk 7]
```diff
@@ -97,19 +109,19 @@ private void setupTemperatureChart() {
...
-        ArrayList<Entry> entries = new ArrayList<>();
-
-        entries.add(new Entry(1, 470));
-        entries.add(new Entry(2, 80));
-        entries.add(new Entry(3, 2300));
-        entries.add(new Entry(4, 1200));
-        entries.add(new Entry(5, 5400));
-        entries.add(new Entry(6, 9100));
-        entries.add(new Entry(7, 2500));
+//        ArrayList<Entry> entries = new ArrayList<>();
```
Comment (grigorebudac) [inline app/app/src/main/java/com/warehouse/ui/dashboard/DashboardFragment.java:112]: remove dead code

**Pull request merged**